### DISCUSSION TITLE
'for in'-Loop

### DISCUSSION THREAD

**Discussion by PhilippElhaus** [dotnet/csharplang#7316] [Language Ideas]
Sometimes you just want to iterate through the properties of an object.
Currently this can only be done with complex Reflection code.

The new `for-in` loop (a mix between a `for` and `foreach` loop) would be a simplification for that.

Example:

```
public class Person
{
    public string Name { get; set; }
    public int Age { get; set; }
    public string Location { get; set; }
}

void forInExample()
{
     Person instance = new();
     for (var prop in instance)
     {
        // Iterates through Public Properties 
        // e.g. Name, Age, Location
     }
}
```

**Comment by En3Tho** [comment]
What is the type of "var" here? Your example doesn't include any real use case also

**Comment by TahirAhmadov** [comment]
This can be easily accomplished with a helper or an extension method. Enumerating properties is a niche scenario, not worth adding to the language.

**Reply by jnm2** [reply]
And it's also a policy-laden scenario with sharp edges; you have protected members, inherited members, shadowing members... no one clear way to set in stone as a `for`...`in` feature.

Perfect for helper methods where specific libraries make domain-specific decisions on this policy.
